### PR TITLE
feat(oidc): OIDC Prompt Support

### PR DESCRIPTION
**Pull request by elimt**
Boundary OIDC method does not currently support passing in prompts during authentication. This change adds the capability of passing OIDC prompts. Prompts are optional OIDC parameters that determine the behaviour of the authentication server: https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest

****NOTE**: This PR will not be merged until the latest release is out**. This is needed to update `boundary`,`boundary/api` & `boundary/sdk` dependencies.

# Changes
Add new optional `prompts` option to `boundary_auth_method_oidc` resource. `prompts` option are optional OIDC parameters that determine the behavior of the authentication server.

## Examples

1. OIDC Auth Method With 1 Prompt

```
resource "boundary_auth_method_oidc" "oidc_with_one_prompt" {
  api_url_prefix     = "https://XO-XO-XO-XO-XOXOXO.boundary.hashicorp.cloud:9200"
  client_id          = "eieio"
  client_secret      = "hvo_secret_XO"
  description        = "My Boundary OIDC Auth Method With Prompt"
  issuer             = "https://sts.windows.net/TENANT_ID/"
  scope_id           = "global"
  signing_algorithms = ["RS256"]
  prompts            = ["select_account"]
  type               = "oidc"
}
```

2. OIDC Auth Method With Multiple Prompts

```
resource "boundary_auth_method_oidc" "oidc_with_multiple_prompts" {
  api_url_prefix     = "https://XO-XO-XO-XO-XOXOXO.boundary.hashicorp.cloud:9200"
  client_id          = "eieio"
  client_secret      = "hvo_secret_XO"
  description        = "My Boundary OIDC Auth Method With Multiple Prompts"
  issuer             = "https://sts.windows.net/TENANT_ID/"
  scope_id           = "global"
  signing_algorithms = ["RS256"]
  prompts            = ["consent", "select_account"]
  type               = "oidc"
}
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,9 +4,9 @@ go 1.21
 
 require (
 	github.com/YakDriver/regexache v0.23.0
-	github.com/hashicorp/boundary v0.13.1-0.20231012004550-1ed0a13004b9
-	github.com/hashicorp/boundary/api v0.0.41
-	github.com/hashicorp/boundary/sdk v0.0.37
+	github.com/hashicorp/boundary v0.14.3-0.20231121173019-2cd2d4ecfceb
+	github.com/hashicorp/boundary/api v0.0.44-0.20231121173019-2cd2d4ecfceb
+	github.com/hashicorp/boundary/sdk v0.0.41-0.20231121173019-2cd2d4ecfceb
```
Comment (elimt) [inline go.mod:9]: **NOTE**: This will be updated with the new release once ready

File: internal/provider/resource_auth_method_oidc.go [hunk 2]
```diff
@@ -536,6 +558,15 @@ func resourceAuthMethodOidcUpdate(ctx context.Context, d *schema.ResourceData, m
 			opts = append(opts, authmethods.WithOidcAuthMethodClaimsScopes(claimsScopes))
 		}
 	}
+	if d.HasChange(authmethodOidcPromptsKey) {
+		if prompts, ok := d.GetOk(authmethodOidcPromptsKey); ok {
+			var promptsList []string
+			for _, alg := range prompts.([]interface{}) {
```
Comment (jimlambrt) [inline internal/provider/resource_auth_method_oidc.go:564]:
```suggestion
			for _, p := range prompts.([]interface{}) {
```

File: internal/provider/resource_auth_method_oidc.go [hunk 3]
```diff
@@ -321,6 +335,14 @@ func resourceAuthMethodOidcCreate(ctx context.Context, d *schema.ResourceData, m
 		opts = append(opts, authmethods.WithOidcAuthMethodClaimsScopes(cList))
 	}
 
+	if prompts, ok := d.GetOk(authmethodOidcPromptsKey); ok {
+		promptList := []string{}
+		for _, c := range prompts.([]interface{}) {
```
Comment (jimlambrt) [inline internal/provider/resource_auth_method_oidc.go:340]: perhaps change `c` to `p`?

File: internal/provider/resource_auth_method_oidc.go [hunk 4]
```diff
@@ -147,6 +148,15 @@ func resourceAuthMethodOidc() *schema.Resource {
 				},
 				Optional: true,
 			},
+			authmethodOidcPromptsKey: {
+				Description: `The prompts passed to the identity provider to determine whether to prompt the end-user for reauthentication, account selection or consent. ` +
```
Comment (jimlambrt) [inline internal/provider/resource_auth_method_oidc.go:152]: perhaps list the valid values?

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,13 @@
 
 Canonical reference for changes, improvements, and bugfixes for the Boundary Terraform provider.
 
+## Next
+
+### New and Improved
+
+* Add support for OIDC prompt. Prompts are optional OIDC parameters that determine the behavior of the authentication server.
```
Comment (jimlambrt) [inline CHANGELOG.md:9]:
```suggestion
* Add support for OIDC prompts. Using prompts, the Relying Party (RP) can customize the authentication and authorization flow to suit their specific needs and improve the user experience. [OIDC Authentication request](https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest)
```

**Review by jimlambrt — CHANGES_REQUESTED** [review]
Looks good.  A few minor suggestions.

**Review by jimlambrt — APPROVED** [review]
ty!

**Pull request merged**